src/utils/ReentrancyGuard.sol

```diff
@@ -8,7 +8,11 @@ abstract contract ReentrancyGuard {
     uint256 private locked = 1;
 
     modifier nonReentrant() virtual {
-        require(locked == 1, "REENTRANCY");
+        // `require(locked == 1)` works fine in normal conditions.
+        // When deployed via proxy, its value would be 0 by default.
+        // Hence, instead of setting the storage slot manually, we can simply
+        // check whether `locked < 2`.
+        require(locked < 2, "REENTRANCY");
```

Comment:
seems more readable to require `locked != 2`

Reply:
iz about gas tho

Reply:
Yeah, ` != 2` increases the gas cost.

![Screenshot 2022-07-12 10:14:29](https://user-images.githubusercontent.com/42379522/178410457-ba8da2cc-7645-4be6-afea-32be86957987.png)